File: src/main/java/musiccube/entities/InstrumentType.java
package musiccube.entities;


import com.fasterxml.jackson.annotation.JsonInclude;

import javax.persistence.*;

@Entity
@JsonInclude(JsonInclude.Include.NON_NULL)
@Table(name = "instrument_type")
public class InstrumentType {

    @Id
    @Column(name = "instrument_type_id")
    @GeneratedValue
    private int id;

    @Column(name = "instrument_type_name", unique = true)
    private String instrumentTypeName;

    @Column(name = "about_type", length = 1024)
    private String aboutType;

    public InstrumentType() {
    }

    public InstrumentType(String typeName) {
        this.instrumentTypeName = typeName;
    }

    public int getId() {
        return id;
    }

    public String getInstrumentTypeName() {
        return instrumentTypeName;
    }

    public String getAboutType() {
        return aboutType;
    }

    public void setInstrumentTypeName(String instrumentTypeName) {
        this.instrumentTypeName = instrumentTypeName;
    }

    public void setAboutType(String aboutType) {
        this.aboutType = aboutType;
    }

}

File: src/app/Class/InstrumentType.ts
export class InstrumentType {

    private id: number;
    private instrumentTypeName: string;
    private aboutType: string;

    constructor(obj?: any) {
        this.id = (obj && obj.id) || 0;
        this.instrumentTypeName = (obj && obj.instrumentTypeName) || "";
        this.aboutType = (obj && obj.aboutType) || "";
    }

    // Getters
    
    public getId(): number {
        return this.id;
    }

    public getInstrumentTypeName(): string {
        return this.instrumentTypeName;
    }

    public getAboutType(): string {
        return this.aboutType;
    }

    // Setters

    public setInstrumentTypeName(instrumentTypeName: string){
        this.instrumentTypeName = instrumentTypeName;
    }

    public setAboutType(aboutType: string) {
        this.aboutType = aboutType;
    }

}